src/Simulation/CsharpGeneration/EntryPoint.fs

```diff
@@ -152,12 +152,15 @@ let private entryPointClass context entryPoint =
     let infoProperty =
         property "Info" (sprintf "EntryPointInfo<%s, %s>" argTypeName returnTypeName)
                         (ident callableName <|.|> ident "Info")
+    let targetIntrinsicProperty =
+        property "TargetIntrinsicsType" "Type" (invoke (ident "typeof") ``(`` [(ident "Microsoft.Quantum.Intrinsic.TargetIntrinsics")] ``)``)
```

Comment:
No matter which target package is being used, it always defines that type, which has the concrete classes used as overrides for the abstract classes generated for body intrinsic callables. However, this is awkward, and thinking about it more, I realized that if I go ahead and update the C# generation to have an option for producing non-abstract classes that call into the interface directly, then the overrides will no longer be needed and I can remove this type parameter and related infrastructure. Since that's the direction we want to go in anyway and it will allow removal of these awkward workarounds, I think it's worth doing as part of this change.

Reply:
With the interface changes worked on by @ScottCarda-MS all of this infrastructure now goes away, and there is no longer a need to infer the originating type for intrinsic overrides :)